### PR TITLE
Rename Layout -> TensorImpl

### PR DESCRIPTION
**Pull request by jainapurva**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #1028

### CODE REVIEW

File: torchao/dtypes/affine_quantized_tensor.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ class AffineQuantizedTensor(TorchAOBaseTensor):
     regardless of the internal representation's type or orientation.
 
     fields:
-      layout_tensor (AQTLayout): tensor that serves as a general layout storage for the quantized data,
+      layout_tensor (AQTTensorImpl): tensor that serves as a general layout storage for the quantized data,
```
Comment (jerryzh168) [inline torchao/dtypes/affine_quantized_tensor.py:134]: rename `layout_tensor` to `tensor_impl` as well?

File: torchao/dtypes/utils.py [hunk 2]
```diff
@@ -4,11 +4,11 @@
 
 """
 Base class for different LayoutType, should not be instantiated directly
```
Comment (jerryzh168) [inline torchao/dtypes/utils.py:6]: LayoutType --> Layout

File: torchao/dtypes/utils.py [hunk 3]
```diff
@@ -4,11 +4,11 @@
...
-used to allow users to pass around configurations for the layout tensor, e.g. inner_k_tiles
-for int4 tensor core tiled layout
+used to allow users to pass around configurations for the tensor impl, e.g. inner_k_tiles
```
Comment (jerryzh168) [inline torchao/dtypes/utils.py:7]:
```suggestion
used to allow users to pass around configurations for the layout, e.g. inner_k_tiles
```

File: torchao/dtypes/utils.py [hunk 4]
```diff
@@ -4,11 +4,11 @@
...
+for int4 tensor core tiled tensor impl
```
Comment (jerryzh168) [inline torchao/dtypes/utils.py:8]:
```suggestion
for int4 tensor core tiled layout
```

File: torchao/dtypes/utils.py (hunk 2 above)
Comment (jainapurva) [reply torchao/dtypes/utils.py:6]: Will add that to another PR in this stack

File: torchao/dtypes/utils.py [hunk 5]
```diff
@@ -4,11 +4,11 @@
...
 
-Note: layout is an abstraction not only for custom data representation, it is also used for how the
-layout interacts with different operators, e.g. the same data representation can have different
+Note: TensorImpl is an abstraction not only for custom data representation, it is also used for how the
```
Comment (jerryzh168) [inline torchao/dtypes/utils.py:10]:
```suggestion
Note: Layout (https://pytorch.org/docs/stable/tensor_attributes.html#torch-layout) is an abstraction not only for custom data representation, it is also used for how the
```

File: torchao/dtypes/utils.py [hunk 6]
```diff
@@ -4,11 +4,11 @@
...
+tensorImpl interacts with different operators, e.g. the same data representation can have different
```
Comment (jerryzh168) [inline torchao/dtypes/utils.py:11]:
```suggestion
Tensor interacts with different operators, e.g. the same data representation can have different
```

File: torchao/dtypes/utils.py [hunk 7]
```diff
@@ -4,11 +4,11 @@
...
 behaviors when running the same operator, e.g. transpose, quantized_linear.
 """
 @dataclass(frozen=True)
```
Comment (jerryzh168) [inline torchao/dtypes/utils.py:14]: also need to rename `LayoutType` to `Layout` as well

File: torchao/utils.py [hunk 8]
```diff
@@ -392,45 +392,45 @@ class MyTensor(torch.Tensor):
     kwarg_types = {k: type(arg) for k, arg in kwargs}
     raise NotImplementedError(f"{cls.__name__} dispatch: attempting to run unimplemented operator/function: {func=}, {types=}, {arg_types=}, {kwarg_types=}")
 
-def _register_layout_cls(cls: Callable, layout_type_class: Callable):
+def _register_layout(cls: Callable, layout_type_class: Callable):
     """Helper function for layout registrations, this is used to implement
-    register_layout_cls decorator for each tensor subclass, see aqt.py for example usage
+    register_layout decorator for each tensor subclass, see aqt.py for example usage
```
Comment (jerryzh168) [inline torchao/utils.py:397]:
```suggestion
    register_layout decorator for each tensor subclass, see affine_quantized_tensor.py for example usage
```

File: torchao/dtypes/utils.py (hunk 7 above)
Comment (jainapurva) [reply torchao/dtypes/utils.py:14]: PR: #1035

**Review by jerryzh168 — APPROVED** [review]
LG, thanks!

**Comment by msaroufim** [comment]
This PR doesn't have a description so it's kind of difficult to know what the intent was here. Was Layout incorrect terminology? Is TensorImpl clearer? What does TensorImpl intend to say?

**Pull request merged**